tests/lib/nested.sh

```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
```

Comment:
Thanks for this!

Reply:
Despite the best intentions. This is a sourced script. It should not `set -e`.

We should either work on making this another tool that is standalone and can set whatever it want or not do this IMO. Instead we should `set -e` in places that use it, although I'm not sure if there is any need for that as it seems to be used in `task.yaml` exclusively.

Reply:
The problem I keep seeing is that for whatever reason the places that source this script (which is as you say mostly task.yaml's) _don't_ have `set -e` set, and so we keep seeing really weird errors because something else further up the chain actually failed and it didn't die here (see the prep-snapd-in-lxd.sh weirdness recently as well). I would rather have this here and have folks complain that it looks awkward but have better test failures specifically because of the following things:

1) the nested tests are already very difficult to debug
2) hunting down why they do not already have `set -e` enabled/set may take a not-insignificant amount of time
3) IMHO _everywhere_ in snapd should be using `set -e` when executed, except for perhaps a few very specific bits of shell script code where we turn it off momentarily then turn it back on. This bit of code helps alleviate that situation

Reply:
Also actually iirc Sergio actually uses this script directly for some things, but I don't know if that's part of master or one of his open PRs the changes to enable that